dstep/translator/MacroDefinition.d

```diff
@@ -0,0 +1,1859 @@
+
+        if (subexpr !is null)
+        {
+            UnaryExpr expr = new UnaryExpr;
+            expr.subexpr = subexpr;
+            expr.operator = spelling;
+            tokens = local;
+            return expr;
+        }
+
+        // FIXME: unary-expression ::= sizeof ( type-name )
+    }
+
+    return parsePostfixExp(tokens, table);
+}
+
+Expression parseCastExpr(ref TokenRange tokens, Cursor[string] table)
+{
+    auto local = tokens;
+
+    if (!accept!("(")(local, TokenKind.punctuation))
+        return parseUnaryExpr(tokens, table);
+
+    Type type = parseTypeName(local, table);
+
+    if (!type.isValid)
+        return parseUnaryExpr(tokens, table);
+
+    if (!accept!(")")(local, TokenKind.punctuation))
+        return parseUnaryExpr(tokens, table);
+
+    auto subexpr = parseCastExpr(local, table);
+
+    if (subexpr is null)
+        return parseUnaryExpr(tokens, table);
+
+    tokens = local;
+
+    CastExpr result = new CastExpr;
+    result.type = type;
+    result.subexpr = subexpr;
+
+    return result;
+}
+
+alias parseMulExpr = parseLeftAssoc!(MulExpr, parseCastExpr, "*", "/", "%");
+alias parseAddExpr = parseLeftAssoc!(AddExpr, parseMulExpr, "+", "-");
+alias parseSftExpr = parseLeftAssoc!(SftExpr, parseAddExpr, "<<", ">>");
+alias parseRelExpr = parseLeftAssoc!(RelExpr, parseSftExpr, "<", ">", "<=", ">=");
+alias parseEqlExpr = parseLeftAssoc!(EqlExpr, parseRelExpr, "==", "!=");
+alias parseAndExpr = parseLeftAssoc!(AndExpr, parseEqlExpr, "&");
+alias parseXorExpr = parseLeftAssoc!(XorExpr, parseAndExpr, "^");
+alias parseOrExpr = parseLeftAssoc!(OrExpr, parseXorExpr, "|");
+alias parseLogicalAndExpr = parseLeftAssoc!(LogicalAndExpr, parseOrExpr, "&&");
+alias parseLogicalOrExpr = parseLeftAssoc!(LogicalOrExpr, parseLogicalAndExpr, "||");
+
+Expression parseCondExpr(ref TokenRange tokens, Cursor[string] table)
+{
+    auto local = tokens;
+
+    Expression expr = parseLogicalOrExpr(local, table);
+
+    if (expr is null)
+        return null;
+
+    tokens = local;
+
+    if (acceptPunctuation!("?")(local))
+    {
+        Expression left = parseExpr(local, table);
+
+        if (left !is null && acceptPunctuation!(":")(local))
+        {
+            Expression right = parseCondExpr(local, table);
+
+            if (right !is null)
+            {
+                CondExpr supexpr = new CondExpr;
+                supexpr.expr = expr;
+                supexpr.left = left;
+                supexpr.right = right;
+                expr = supexpr;
+
+                tokens = local;
+            }
+        }
+    }
+
+    return expr;
+}
+
+bool parseBasicSpecifier(ref TokenRange tokens, ref string spelling, Cursor[string] table)
+{
+    import std.meta : AliasSeq;
+
+    alias specifiers = AliasSeq!(
+        "void",
+        "char",
+        "short",
+        "int",
+        "long",
+        "float",
+        "double",
+        "signed",
+        "unsigned",
+        // "__complex__", TBD
+        // "_Complex", TBD
```

Comment:
What's the status of these two "complex", do you plan to implement them in this PR?